sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamIOSourceRel.java

```diff
@@ -94,7 +116,19 @@ public NodeStats estimateNodeStats(RelMetadataQuery mq) {
           "Should not have received input for %s: %s",
           BeamIOSourceRel.class.getSimpleName(),
           input);
-      return beamTable.buildIOReader(input.getPipeline().begin());
+
+      PBegin begin = input.getPipeline().begin();
+      BeamSqlTableFilter filters = beamTable.constructFilter(ImmutableList.of());
+
+      if (usedFields.isEmpty() && filters instanceof DefaultTableFilter) {
+        return beamTable.buildIOReader(begin);
+      }
+
+      Schema newBeamSchema = CalciteUtils.toSchema(getRowType());
+
+      return beamTable
+          .buildIOReader(input.getPipeline().begin(), filters, usedFields)
```

Comment:
Replace `input.getPipeline().begin()` with `begin` (or drop the variable).

Reply:
Nice catch! Updated to use local variable `begin`.